docs/DeveloperGuide.md

```diff
@@ -1,34 +1,579 @@
+
+**Step 3.** The `LogicManager` calls `AddCommand#execute`, which calls `TaskList#addTask` to create a task based on the given parameters, and adds the task to the list. The `TaskList` now contains 1 task (Assignment1). The message to show if the task is added successfully is subsequently outputted by the `AthenaUi` class to the user.
+
+![AddTaskEmptyListObjectDiagram](objectDiagrams/addTask/onetask.png)
+
+**Step 4.** After the command is executed, `LogicManager` calls `Storage#saveTaskListData` to automatically save the tasks in the `TaskList` into the save file.
+
+**Step 5.** Upon completion of execution, `LogicManager` returns a boolean value `false` to `Athena` to allow the continuous run of the program. 
+
+The following sequence diagram illustrates how Step 3 of the task adding operation works:
+
+![AddTaskSequenceDiagram](sequenceDiagrams/AddCommand%20diagram.png)
+
+### Delete task feature
+The deleting task mechanism is facilitated by `LogicManager`.
+
+The retrieving of input details is done by `Parser`. It splits the user input based on the command type and the index of task.
+The `DeleteCommand` class will then be executed and the task will be removed from the `TaskList`.
+
+`Parser`, `DeleteCommand`, `TaskList` implements the following operations:
+
+* `Parser#parse` - Parse user input to retrieve the command type and index of task.
+* `Parser#parseDeleteCommand` - Parse user input to retrieve the index of task to be deleted and creates an `DeleteCommand` object.
+* `DeleteCommand#execute` - Remove task into `TaskList` and calls `Ui` to print message output.
+* `Storage#saveTaskListData` - Updates the current task list into the save file.
+
+Given below is an example usage scenario and how the deleting task mechanism behaves at each step.
+
+**Step 1.** The user deletes a task from the application, by inputting `delete 1`. 
+
+**Step 2.** The input will be read in by the `Athena` class. The input will be passed into `LogicManager` where `Parser#parse` will parse the user input to get the command type and details then call`Parse#parseDeleteCommand`to create a `DeleteCommand` object.The `DeleteCommand` object is returned to the `LogicManager`.
+
+**Step 3.** The `LogicManager` calls `DeleteCommand#execute`. The `TaskList` now removed task with index 1. The message to show if the task is deleted successfully is subsequently outputted by the `AthenaUi` class to the user.
+
+**Step 4.** After the command is executed, `LogicManager` calls `Storage#saveTaskListData` to automatically save the tasks in the `TaskList` into the save file.
+
+**Step 4.** Upon completion of execution, `LogicManager` returns a boolean value `false` to `Athena` to allow the continuous run of the program. 
+
+The following sequence diagram illustrates how Step 3 of the task deleting operation works:
+
+![DeleteTaskSequenceDiagram](sequenceDiagrams/DeleteCommand%20diagram.png)
```

Comment:
Is getTaskRestore() call put wrongly under DeleteCommand as opposed to under TaskList? Since the deleteTask() has not returned to DeleteCommand it seems wrong to have another function call, not to mention the call returns to deleteTask() instead.